plugins/search-backend/src/service/router.test.ts

```diff
@@ -208,5 +211,55 @@ describe('createRouter', () => {
         });
       });
     });
+
+    describe('Be less restrictive with unknown keys on query endpoint', () => {
+      const expectedResult = {
+        type: 'software-catalog',
+        document: {
+          text: 'foo',
+          title: 'bar baz',
+          location: '/catalog/default/component/example',
+        },
+      };
+      beforeAll(async () => {
```

Comment:
If you're able to switch to just asserting on the query arguments, you could probably remove a lot of this test setup code too.

You could even consider forgoing this whole `describe()` block, and just adding the `pass through unknown keys` case into one of the existing describe blocks.  Perhaps `GET /query` block?